test: bsim: Notification EATT test

CarbuneanuAlexandru opened the pull request:
Increasing the test coverage of notification on EATT bearers.

Signed-off-by: Alexandru Carbuneanu <alexandru.carbuneanu@nordicsemi.no>

hermabe committed:
Bluetooth: Host: Automatically create EATT channels on connection

If the Kconfig BT_EATT_AUTO_CONNECT is enabled (on by default), the
host will try to connect BT_EATT_MAX EATT channels when a connection
to a peer is established.

Signed-off-by: Herman Berget <herman.berget@nordicsemi.no>

hermabe committed:
Bluetooth: Host: Make bt_eatt_connect connect up to BT_EATT_MAX channels

bt_eatt_connect may now send multiple credit based connection requests.
Previously it would only send one and thus only allow 5 EATT channels to
be connected.

Signed-off-by: Herman Berget <herman.berget@nordicsemi.no>

hermabe committed:
Bluetooth: Host: Make bt_eatt_connect public

This makes it possible for applications to create Enhanced ATT channels.

Signed-off-by: Herman Berget <herman.berget@nordicsemi.no>

hermabe committed:
Bluetooth: Host: Add EATT channel callback

The application can register a callback to get notified when an EATT
channel is connected.

Signed-off-by: Herman Berget <herman.berget@nordicsemi.no>

hermabe committed:
Bluetooth: Host: Cancel EATT connection work in att_reset

When att->conn is set to NULL in att_reset, EATT connection will fail.

Signed-off-by: Herman Berget <herman.berget@nordicsemi.no>

hermabe committed:
Bluetooth: Host: Update connection test to use EATT channel callback

The new API lets the application know when EATT channels are connected,
so it does not need to sleep 10 seconds.

Signed-off-by: Herman Berget <herman.berget@nordicsemi.no>

jhedberg reviewed (changes requested):
This PR needs to be split up into multiple commits. At the very least the stack changes (including public APIs) should be in a separate commit.

alwa-nordic commented on include/bluetooth/att.h:
```diff
@@ -65,6 +65,34 @@ size_t bt_eatt_count(struct bt_conn *conn);
 #endif /* CONFIG_BT_TESTING */
 #endif /* CONFIG_BT_EATT */
 
+<<<<<<< HEAD
```
Bad merge.

CarbuneanuAlexandru committed:
test: bsim: Notification EATT test

Increasing the test coverage of notification on EATT bearers.
- test implementation

Signed-off-by: Alexandru Carbuneanu <alexandru.carbuneanu@nordicsemi.no>

Thalley commented:
Looks like this builds on top of https://github.com/zephyrproject-rtos/zephyr/pull/43112 - In the future please do add reference to other PRs that it may build on, so it is clear for the reviewers which commits to review. 

Alternatively, only upload the changes you want reviewed, and have the PR fail on CI. 

You can also create PRs as `draft` so that you can run your CI etc., but without triggering any review requests :)

I'll review this PR once https://github.com/zephyrproject-rtos/zephyr/pull/43112 is merged.

CarbuneanuAlexandru committed:
Bluetooth: Host: Callback function for EATT disconnection

Increasing the test coverage of notification on EATT bearers.
- callback function for disconnecting EATT bearers

Signed-off-by: Alexandru Carbuneanu <alexandru.carbuneanu@nordicsemi.no>

CarbuneanuAlexandru commented:
> Looks like this builds on top of #43112 - In the future please do add reference to other PRs that it may build on, so it is clear for the reviewers which commits to review.
> 
> Alternatively, only upload the changes you want reviewed, and have the PR fail on CI.
> 
> You can also create PRs as `draft` so that you can run your CI etc., but without triggering any review requests :)
> 
> I'll review this PR once #43112 is merged.

yes, I was not sure how to handle it, it needs changes from an other PR to work, so the commits to be reviewed here are only the last two once:
d1082a2160e2d3637348d3ba71163ab01b42121d
a9b46352389e2643a2dfb42eb1b6a2e31355d5b8
